jobs/furnishings/src/furnishings/stage_processors/stage_two.py

```diff
@@ -0,0 +1,68 @@
+# Copyright © 2024 Province of British Columbia
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Furnishings job procssing rules for stage two of involuntary dissolution."""
+from datetime import datetime
+
+from flask import Flask
+from legal_api.models import Batch, BatchProcessing, Business, Furnishing, db
+from sqlalchemy import exists, not_
+
+
+def process(app: Flask):
+    """Run process to manage and track notifications for dissolution stage two process."""
+    try:
+        furnishing_subquery = exists().where(
+            Furnishing.batch_id == BatchProcessing.batch_id,
+            Furnishing.business_id == BatchProcessing.business_id,
+            Furnishing.furnishing_name.in_([
+                Furnishing.FurnishingName.INTENT_TO_DISSOLVE,
+                Furnishing.FurnishingName.INTENT_TO_DISSOLVE_XPRO
+            ])
+        )
+        batch_processings = (
+            db.session.query(BatchProcessing)
+            .filter(BatchProcessing.status == BatchProcessing.BatchProcessingStatus.PROCESSING)
+            .filter(BatchProcessing.step == BatchProcessing.BatchProcessingStep.WARNING_LEVEL_2)
+            .filter(Batch.id == BatchProcessing.batch_id)
+            .filter(Batch.batch_type == Batch.BatchType.INVOLUNTARY_DISSOLUTION)
+            .filter(Batch.status == Batch.BatchStatus.PROCESSING)
+            .filter(not_(furnishing_subquery))
+        ).all()
+
+        grouping_identifier = Furnishing.get_next_grouping_identifier()
```

Comment:
Can you remind me what we use this grouping_identifier for?

Reply:
As far as I know, it will be used for tracking purpose when we send paper letters, generate data files to BC Laws. 🤔

Reply:
The way I envisioned the `grouping_identifier` working is it provides a way of tracking which furnishing entries went out for a specific furnishing run grouped by furnishing type.  Furnishing entries belong to furnishing type can also span multiple batches so this will help track across batches too.

Initially I was thinking of just using this grouping_identifer field for just letters and gazette entries.  But I'm thinking we just use it across all furnishing types now as it costs us nothing really to add it.